src/commands/signzone.rs

```diff
@@ -383,7 +383,7 @@ impl SignZone {
         res.map_err(|err| Error::from(format!("Invalid timestamp: {err}")))
     }
 
-    pub fn execute(self, env: impl Env) -> Result<(), Error> {
+    pub fn execute(self, env: impl Env, is_ldns: bool) -> Result<(), Error> {
```

Comment:
An alternative to this is to add a field to `Signzone` with `is_ldns` which is set to `false` by clap and `true` at the end of `parse_ldns`. That seems about as convenient, but also doesn't affect the other utils.

Reply:
I think developing better interfaces and functionality in the `dnst` versions of the commands we have will eventually necessitate this kind of parameter in more places.  We don't need to do that immediately, but I think it's something to keep in mind.

Reply:
I was aware of your idea to do it the other way @tertsdiepraam, but as I said I prefer it this way as I think it should be implemented once in a common single mechanism for all commands, not separately per command with potentially divergent implementations to solve a common problem.

Reply:
Sure! I made my case and stand by it, but it's alright! You've got a very practical solution here.

Reply:
I can live with going with the majority opinion in the team, maybe others don't agree with my stated reasons and prefer your approach. It's easier to choose among actual implementations than hypothetical designs so now we have both and can pick one.